### PR TITLE
Update styling components theme

### LINKED ISSUE #383: Redo approach to use styled-components themes in gearbox components

### LINKED ISSUE #394: TimeseriesChartMeta redo theme approach

### LINKED ISSUE #395: ThreeDNodeTree redo theme approach

### PR DESCRIPTION
**Pull request by joyrostie**
Created:
* HOC WithDefaultTheme to wrap component and compose themes **mrtraser** 
* Stories WithTheme for affected components 
* Unit test for *ThreeDNodeTree*

Modified: 
* AssetTree
* EventPreview
* TenantSelector
* ThreeDNodeTree
* TimeseriesChartMeta

Removed:
* *ThemeProvider*, now using bundled with *styled-components*

Closes #383, closes #394, closes #395

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by mrtraser** [comment]
Guess we need to apply **withDefaultTheme** HOC to all exported components

**Comment by joyrostie** [comment]
> Guess we need to apply **withDefaultTheme** HOC to all exported components

WIll do.

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

File: src/components/common/RootAssetSelect/RootAssetSelect.test.tsx [hunk 1]
```diff
@@ -28,21 +32,29 @@ describe('RootAssetSelect', () => {
   });
 
   it('onSelectAsset should be triggered', async done => {
```
Comment (joyrostie) [inline src/components/common/RootAssetSelect/RootAssetSelect.test.tsx]: @mrtraser Can you please check this test. Looks like there is some issue to emulate onChange event after finding the Select element. I've tried on both wrapped and pure component exports, different instances, spy functions but onSelectAsset is not changing component's state.

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

File: src/hoc/withDefaultTheme.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            >)}
+            key=""
+          />
+        </WithThemeComponent>
+      );
+    }
+  };
+};
+
+// export function withDefaultTheme<P>(WrapperComponent: React.ComponentType<P>) {
```
Comment (khalzmax) [inline src/hoc/withDefaultTheme.tsx]: Delete commented block if it's not needed

File: src/components/AssetEventsPanel/components/AssetEventsPanelPure.tsx [hunk 3]
```diff
@@ -236,3 +248,8 @@ const EventMetadataList = styled.div`
     }
   }
 `;
+
+const Component = withDefaultTheme(AssetEventsPanelPureComponent);
```
[the remark this answers is not in the document]
Comment (joyrostie) [reply src/components/AssetEventsPanel/components/AssetEventsPanelPure.tsx]: Actually, it's vice versa, AssetEventsPanelPureComponent is the component without theme wrapping, while AssetEventsPanelPure is wrapped with WithDefaultTheme under its original name.

File: src/components/common/RootAssetSelect/RootAssetSelect.test.tsx [hunk 4]
```diff
@@ -28,24 +32,37 @@ describe('RootAssetSelect', () => {
...
+    const liElement = wrapper.find('.ant-select-dropdown-menu-item-selected');
+
+    expect(onSelectAsset).toHaveBeenCalled();
+    expect(onAssetSelected).toHaveBeenCalled();
+    expect(liElement.text()).toBe(assetsList[0].description);
+    expect(wrapper.find(RootAssetSelectComponent).state('current')).toEqual(
+      assetId
+    );
+
+    done();
```
[the remark this answers is not in the document]
Comment (joyrostie) [reply src/components/common/RootAssetSelect/RootAssetSelect.test.tsx]: Okay, I will refactor the test.

File: src/hoc/withDefaultTheme.tsx (hunk 2 above)
Comment (joyrostie) [reply src/hoc/withDefaultTheme.tsx]: Good notice :-)

File: src/components/common/RootAssetSelect/RootAssetSelect.test.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (joyrostie) [reply src/components/common/RootAssetSelect/RootAssetSelect.test.tsx]: Fixed.

File: src/hoc/withDefaultTheme.tsx (hunk 2 above)
Comment (joyrostie) [reply src/hoc/withDefaultTheme.tsx]: Done.

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-399.gearboxjs.preview.cogniteapp.com

**Pull request merged**